src/Planner/PlannerJoinTree.h

```diff
@@ -11,10 +11,13 @@
 namespace DB
 {
 
+using ColumnIdentifierSet = std::unordered_set<ColumnIdentifier>;
```

Comment:
Need to move this into `TableExpressionData` and reuse in `CollectColumnIdentifiers`.